src/Workspaces/Core/Desktop/Workspace/Host/Documentation/DocumentationProviderServiceFactory.cs

```diff
@@ -29,16 +30,53 @@ public DocumentationProvider GetDocumentationProvider(string assemblyPath)
                     throw new ArgumentNullException(nameof(assemblyPath));
                 }
 
-                assemblyPath = Path.ChangeExtension(assemblyPath, "xml");
+                string xmlDocumentationFilePath;
+                if (!TryFindDocumentationFile(assemblyPath, out xmlDocumentationFilePath))
+                {
+                    return DocumentationProvider.Default;
+                }
 
                 DocumentationProvider provider;
-                if (!_assemblyPathToDocumentationProviderMap.TryGetValue(assemblyPath, out provider))
+                if (!_assemblyPathToDocumentationProviderMap.TryGetValue(xmlDocumentationFilePath, out provider))
                 {
-                    provider = _assemblyPathToDocumentationProviderMap.GetOrAdd(assemblyPath, _path => new FileBasedXmlDocumentationProvider(_path));
+                    provider = _assemblyPathToDocumentationProviderMap.GetOrAdd(xmlDocumentationFilePath, _path => new FileBasedXmlDocumentationProvider(_path));
                 }
 
                 return provider;
             }
+
+            private bool TryFindDocumentationFile(string assemblyPath, out string xmlDocumentationFilePath)
+            {
+                var xmlFilePath = string.Empty;
+                var xmlFileName = Path.ChangeExtension(Path.GetFileName(assemblyPath), "xml");
+                var originalDirectory = Path.GetDirectoryName(assemblyPath);
+
+                // 1. Look in subdirectories based on current culture
+                var culture = CultureInfo.CurrentCulture;
```

Comment:
Should this be `CurrentCulture`, or `CurrentUICulture`?  Also, can this be called from a background thread?  If so, is CurrentCulture properly defined there?

Tagging @davkean.